### DISCUSSION TITLE
usage of stock removal API?

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#7187] [Q&A]
I've got a python script to remove stock from a part which works most of the time. I've been trying to track down a stock removal issue on my script which I think I found the cause of. 

I don't think I understand the stock removal process correctly or I'm missing something. 

To use the API to remove stock you need to specify the amount of stock to remove and the StockItem number.

My script does a API request for example for part 911:

```
http://demo.inventree.org/api/stock/?format=json&part=911
```

This then returns the PK number for the stock item where I then call
```
http://demo.inventree.org/api/stock/remove
``` 
and specify the PK and Quantity to remove.  This works 100% for a part number like 82 on the demo site (Which has a single stock item)

The problem I have is when there are other stock items listed, like "consumed by build order'

![image](https://github.com/inventree/InvenTree/assets/1429043/21302ad0-1ac8-43e4-9d92-281ae9cf5aaf)

Programmatically how do I know I am supposed to remove stock from PK 1223, rather than 1227 as both are returned?

![image](https://github.com/inventree/InvenTree/assets/1429043/693d6c22-794e-4d13-8fa2-1ba6b238667b)

**Comment by StormTrooper** [comment]
I think I found the answer. If I use &in_stock=1 it seems to solve one issue, but that  leads me to another issue.

If I have 2 stock items each with say 5 pieces, and I call the remove stock API for quantity of 6.  How do I handle this?

Remove 5 from one of the stock items and 1 from the other stock item? That's going to get messy then as I then first have to query the quantity of each stock item and work out how much to remove and how to split the removal across all the stock items?

There must be a another way I'm not aware of?

**Reply by SchrodingersGat** [reply]
The complexity matches the physical reality - if you have two distinct stock items, you need to specify where you are removing from. You could merge the two items together, into a single stock item, if you like.

**Reply by StormTrooper** [reply]
Thanks. I'll try out the merge option. 
Failing that I'll retrieve all stock items and build in some logic to delete from multiple if one of them does not have enough quantity. 

Just a query on the stock/remove API call - if I try and remove say 4 pieces and there are only 2 shouldn't it give some kind of API error, or in the response show the actual value removed.  

I  have 2 stock items with 2 pieces in each. I then try to remove 4 from one of them I get:

```
{"items":[{"pk":null,"quantity":"4.00000","status":null}]}
```

Should it report 2 were removed?
